packages/mdc-menu-surface/README.md

```diff
@@ -0,0 +1,164 @@
+`isRtl() => boolean` | Returns boolean indicating whether the current environment is RTL.
+`setTransformOrigin(value: string) => void` | Sets the transform origin for the menu surface element.
+`focus() => void` | Focuses the root element of the menu surface.
+`isFocused() => boolean` | Returns a boolean value indicating whether the root element of the menu surface is focused.
+`saveFocus() => void` | Stores the currently focused element on the document, for restoring with `restoreFocus`.
+`restoreFocus() => void` | Restores the previously saved focus state, by making the previously focused element the active focus again.
+`isFirstElementFocused() => boolean` | Returns a boolean value indicating if the first focusable element of the menu-surface is focused.
+`isLastElementFocused() => boolean` | Returns a boolean value indicating if the last focusable element of the menu-surface is focused.
+`focusFirstElement() => void` | Focuses the first focusable element of the menu-surface.
+`focusLastElement() => void` | Focuses the last focusable element of the menu-surface.
+`getInnerDimensions() => {width: number, height: number}` | Returns an object with the items container width and height.
+`getAnchorDimensions() => {width: number, height: number, top: number, right: number, bottom: number, left: number}` | Returns an object with the dimensions and position of the anchor (same semantics as `DOMRect`).
+`getBodyDimensions() => {width: number, height: number}` | Returns an object with width and height of the body, in pixels.
+`getWindowDimensions() => {width: number, height: number}` | Returns an object with width and height of the viewport, in pixels.
+`getWindowScroll() => {x: number, y: number}` | Returns an object with the amount the body has been scrolled on the `x` and `y` axis.
+`setPosition(position: {top: string, right: string, bottom: string, left: string}) => void` | Sets the position of the menu surface element.
+`setMaxHeight(value: string) => void` | Sets `max-height` style for the menu surface element.
+
+### `MDCMenuSurfaceFoundation`
+
+Method Signature | Description
+--- | ---
+`setAnchorCorner(corder: Corner) => void` | Sets the corner that the menu surface will be anchored to. See [constants.js](./constants.js)
+`setAnchorMargin(margin: AnchorMargin) => void` | Sets the distance from the anchor point that the menu surface should be shown.
+`setIsHoisted(isHoisted: boolean) => void` | Sets whether the menu surface has been hoisted to the body so that the offsets are calculated relative to the page and not the anchor.
+`setFixedPosition(isFixed: boolean) => void` | Sets whether the menu surface is using fixed positioning.
+`setAbsolutePosition(x: number, y: numnber) => void` | Sets the absolute x/y position of the menu. Should only be used when the menu is hoisted or using fixed positioning.
+`handleDocumentClick(event: Event) => void` | Method used as the callback function for the `click` event.
```

Comment:
Should this be `handleBodyClick`, or should the event [de]registration APIs actually use `document.documentElement`, not `document.body` (in case the body doesn't span the full viewport)?